### PR TITLE
Update video_id regexp to account for recent abc.com changes (fixes #25197 and #25207)

### LINKED ISSUE #25197: [Go] Abc.com giving Adobe Pass error
## Checklist

- [x] I'm reporting a broken site support
- [x] I've verified that I'm running youtube-dl version **2020.05.08**
- [x] I've checked that all provided URLs are alive and playable in a browser
- [x] I've checked that all URLs and arguments with special characters are properly quoted or escaped
- [x] I've searched the bugtracker for similar issues including closed ones

## Verbose log

```
$ /usr/local/bin/youtube-dl -v https://abc.com/shows/general-hospital/episode-guide/2020-04/30-general-hospital-43020
[debug] System config: []
[debug] User config: []
[debug] Custom config: []
[debug] Command-line args: [u'-v', u'https://abc.com/shows/general-hospital/episode-guide/2020-04/30-general-hospital-43020']
[debug] Encodings: locale UTF-8, fs UTF-8, out UTF-8, pref UTF-8
[debug] youtube-dl version 2020.05.08
[debug] Python version 2.7.18rc1 (CPython) - Linux-5.4.0-29-generic-x86_64-with-Ubuntu-20.04-focal
[debug] exe versions: ffmpeg 4.2.2, ffprobe 4.2.2, rtmpdump 2.4
[debug] Proxy map: {}
[Go] 30-general-hospital-43020: Downloading webpage
[Go] VDKA0_uixgx8ru: Downloading JSON metadata
ERROR: This video is only available for users of participating TV providers. Use --ap-mso to specify Adobe Pass Multiple-system operator Identifier and --ap-username and --ap-password or --netrc to provide account credentials.
Traceback (most recent call last):
  File "/usr/local/bin/youtube-dl/youtube_dl/YoutubeDL.py", line 797, in extract_info
    ie_result = ie.extract(url)
  File "/usr/local/bin/youtube-dl/youtube_dl/extractor/common.py", line 530, in extract
    ie_result = self._real_extract(url)
  File "/usr/local/bin/youtube-dl/youtube_dl/extractor/go.py", line 187, in _real_extract
    url, video_id, requestor_id, resource)
  File "/usr/local/bin/youtube-dl/youtube_dl/extractor/adobepass.py", line 1414, in _extract_mvpd_auth
    raise_mvpd_required()
  File "/usr/local/bin/youtube-dl/youtube_dl/extractor/adobepass.py", line 1379, in raise_mvpd_required
    'and --ap-username and --ap-password or --netrc to provide account credentials.', expected=True)
ExtractorError: This video is only available for users of participating TV providers. Use --ap-mso to specify Adobe Pass Multiple-system operator Identifier and --ap-username and --ap-password or --netrc to provide account credentials.
```

## Description

I regularly download General Hospital from Abc.com.  This worked fine on 5/6/20, but failed yesterday.  I updated to the 5/8/20 release of youtube-dl, but that didn't fix the problem.
Thanks!

### LINKED ISSUE #25207: ABC [GO]: Unable to retrieve required files for playback.
## Checklist

- [x] I'm reporting a broken site support
- [x] I've verified that I'm running youtube-dl version **2020.05.08**
- [x] I've checked that all provided URLs are alive and playable in a browser
- [x] I've checked that all URLs and arguments with special characters are properly quoted or escaped
- [x] I've searched the bugtracker for similar issues including closed ones

## Verbose log

```
$ youtube-dl --version
2020.05.08

$ youtube-dl https://abc.com/shows/the-goldbergs/episode-guide/season-07/22-the-fakeup -v
[debug] System config: []
[debug] User config: [u'--download-archive', u'/home/pat/archive-youtube-dl.txt', u'--retries', u'infinite', u'--fragment-retries', u'infinite', u'--abort-on-unavailable-fragment']
[debug] Custom config: []
[debug] Command-line args: [u'https://abc.com/shows/the-goldbergs/episode-guide/season-07/22-the-fakeup', u'-v']
[debug] Encodings: locale UTF-8, fs UTF-8, out UTF-8, pref UTF-8
[debug] youtube-dl version 2020.05.08
[debug] Python version 2.7.17 (CPython) - Linux-4.15.0-96-generic-x86_64-with-Ubuntu-18.04-bionic
[debug] exe versions: ffmpeg N-95880-gd07d1c1, ffprobe N-95880-gd07d1c1, phantomjs 2.1.1, rtmpdump 2.4
[debug] Proxy map: {}
[Go] 22-the-fakeup: Downloading webpage
[Go] VDKA0_uixgx8ru: Downloading JSON metadata
[Go] VDKA0_uixgx8ru: Retrieving Media Token
[Go] VDKA0_uixgx8ru: Downloading JSON metadata
ERROR: Go said: 1025:Unable to retrieve required files for playback. Please try again in a few minutes.
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/youtube_dl/YoutubeDL.py", line 797, in extract_info
    ie_result = ie.extract(url)
  File "/usr/local/lib/python2.7/dist-packages/youtube_dl/extractor/common.py", line 530, in extract
    ie_result = self._real_extract(url)
  File "/usr/local/lib/python2.7/dist-packages/youtube_dl/extractor/go.py", line 205, in _real_extract
    raise ExtractorError('%s said: %s' % (self.IE_NAME, error_message), expected=True)
ExtractorError: Go said: 1025:Unable to retrieve required files for playback. Please try again in a few minutes.
```

## Description

These URL's were working fine just a few weeks ago.  This change seems to have happened recently?

### PR DESCRIPTION
**Pull request by tmthywynn8**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [adding new extractor tutorial](https://github.com/ytdl-org/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/ytdl-org/youtube-dl#youtube-dl-coding-conventions) sections
- [x] [Searched](https://github.com/ytdl-org/youtube-dl/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [ ] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [ ] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [x] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Bug fix
- [ ] Improvement
- [ ] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

This pull request fixes #25197 and #25207 by accounting for the new changes on abc.com&mdash;first noticed on Wednesday, May 6. Now, the unique video ID can be found by searching for videoIdCode; note that only one instance will be found), potentially avoiding any false positives, e.g., other video IDs referenced on the page.

The solution was to change the regexp on line 141 and simplifying it for the new name on the video's source. The Rookie video in the examples dictionary can be used to verify the new changes.

### CODE REVIEW

**Comment by magnum32** [comment]
> I verified this fix allows downloading from abc.com. I reviewed the code change, it looks fine to me.

i updated YouTube DLG and it still doesn't work for some reason

**Comment by cliss** [comment]
Bump; there have been [two](https://github.com/ytdl-org/youtube-dl/releases/tag/2020.05.29) [releases](https://github.com/ytdl-org/youtube-dl/releases/tag/2020.06.06) since this PR was opened; any chance we can see it land in the next release, please? 😇

File: youtube_dl/extractor/go.py [hunk 1]
```diff
@@ -138,7 +138,7 @@ def _real_extract(self, url):
                     # from http://freeform.go.com/shows/shadowhunters/episodes/season-2/1-this-guilty-blood
                     r'data-video-id=["\']*(VDKA\w+)',
                     # https://abc.com/shows/the-rookie/episode-guide/season-02/03-the-bet
-                    r'\b(?:video)?id["\']\s*:\s*["\'](VDKA\w+)'
```
Comment (dstftw) [inline youtube_dl/extractor/go.py:141]: Do not touch old patterns.

**Comment by magnum32** [comment]
the fix still has not been added to youtube-dl. what is going on?

**Comment by DarF234** [comment]
May I respectfully inquire as to what is delaying this fix?  ABC is a very popular site and has the habit of reducing retention of free _and_ logged-in content at seemingly random times.  A timely fix would be most welcome.  I am assuming (possibly incorrectly) that dstftw's request is the one we are waiting on, as he has been handling this issue.
All the patches I've found don't correspond with the code.  So the only solution seems to be here.
I normally have high praise for the rapid updates to your software.  But I believe this is the longest period of time I have seen without an update.

**Comment by chrisallen** [comment]
Could we please get this merged?

**Comment by DarF234** [comment]
The problem has changed.  I was stupid enough to not check logs until today.  I'm no longer getting the Adobe Pass error.  Now I'm getting "ExtractorError: Go said: 1025:Unable to retrieve required files for playback. Please try again in a few minutes."
An error that was assumed to be fixed in early may.  This probably explained why the program worked that one evening.  So the latest version may not have the Adobe Pass error, but it definitely has the one above.  I won't be able to post an issue for more than 16 hours.  If someone reading this can, please run a test on any **full** ABC show (clips download okay) that's not locked.  If you get error 1025, if there's nothing open on it already, please post an issue ASAP!  You'll be doing us all a favor by getting this error on the right track.
It it is an open issue, please provide a URL to it so we can follow it.

File: youtube_dl/extractor/go.py (hunk 1 above)
Comment (tony-izzo) [reply youtube_dl/extractor/go.py:141]: Sorry for not minding my own business, but it looks like your requested change was implemented by @tmthywynn8 a couple of months ago.  I'm not that familiar with the GitHub workflow, but it looks like this is just waiting for your approval.

Comment (tmthywynn8) [reply youtube_dl/extractor/go.py:141]: Yes. I made the necessary changes in commit 783d296 and requested another review to make sure that I fulfilled the requirements. I'm not sure what the procedure is suppose to be, so maybe requesting another review was the wrong thing to do

Comment (tony-izzo) [reply youtube_dl/extractor/go.py:141]: I'm not sure yet but I think removing the original pattern (by replacing it with the new pattern) fixed the problem, and putting it back has re-broken it.  My suspicion is that both patterns are matched on the page in question but one returns an invalid ID.

If I'm right, undoing the requested change will get it working again.  But that's not really a solution, and again I'm not sure yet.

Comment (tony-izzo) [reply youtube_dl/extractor/go.py:141]: Sorry it took me so long to get to this, but I just tested #25216 and it works for me as-is.  So my theory above is completely wrong.

**Comment by kucksdorfs** [comment]
What's going on with this PR? This has been implemented in youtube-dlc months ago: https://github.com/blackjack4494/youtube-dlc/pull/149#event-3816849149

**Comment by llacb47** [comment]
It will not be merged until maintainers fly to US, purchase a cable package, and test it themselves. And no, I'm not joking.

**Comment by kucksdorfs** [comment]
> 
> 
> It will not be merged until maintainers fly to US, purchase a cable package, and test it themselves. And no, I'm not joking (well maybe a little bit).

Would SSH access to a Raspberry Pi on a US ISP work?

**Comment by rautamiekka** [comment]
> > It will not be merged until maintainers fly to US, purchase a cable package, and test it themselves. And no, I'm not joking (well maybe a little bit).
> 
> Would SSH access to a Raspberry Pi on a US ISP work?

Don't you need to play the vids too ?

**Comment by kucksdorfs** [comment]
> > > It will not be merged until maintainers fly to US, purchase a cable package, and test it themselves. And no, I'm not joking (well maybe a little bit).
> > 
> > 
> > Would SSH access to a Raspberry Pi on a US ISP work?
> 
> Don't you need to play the vids too ?

I think this could be handled by rsync or X11 forwarding (depending on how testing would need to get done). Or you know, whoever provides the SSH endpoint could also make sure the mp4 (or whatever) file could make sure the file is playable.

**Comment by owenthewizard** [comment]
I have a subscription and would be willing to provide any needed access to the maintainers. I'm located in the US.

**Comment by kucksdorfs** [comment]
I don't have the login, but I also have a residential IP address in the United States.

**Pull request closed without merging**